guides/authentication/mix_phx_gen_auth.md

```diff
@@ -0,0 +1,87 @@
+
+## Getting Started
+
+Let's start by running the following command from the root of our app (or `apps/my_app_web` in an umbrella app):
+
+    $ mix phx.gen.auth Accounts User users
+
+This creates an `Accounts` context with an `Accounts.User` schema module. The final argument is the plural version of the schema module which is used for generating database table names and route helpers. The `mix phx.gen.auth` generator is similar to `mix phx.gen.html` except it does not accept a list of additional fields to add to the schema and it generates many more context functions.
+
+Since this generator installed additional dependencies in `mix.exs`, let's fetch those dependencies:
+
+    $ mix deps.get
+
+Now we need to verify the database connection details for the development and test environments in `config/` so the migrator and tests can run properly. Then run the following to create the database:
+
+    $ mix ecto.setup
+
+Let's run the tests to make sure our new authentication system works as expected.
+
+    $ mix test
+
+And finally, let's start our phoenix server and try it out.
+
+    $ mix phx.server
+
+## Developer Responsibilities
+
+Since phoenix generates this code into your application instead of building these modules into phoenix itself, you now have complete freedom to modify the authentication system so it works best with your use case. The one caveat with using a generated authentication system is it will not be updated after it's been generated. Therefore as improvements are made to the output of `mix phx.gen.auth`, it becomes your responsibility to determine if these changes need to be ported into your application.
+
+## Generated Code
+
+The following are notes about the generated authentication system.
+
+### Password hashing
+
+The password hashing mechanism defaults to `bcrypt` for Unix systems and `pbkdf2` for Windows systems. Both systems use [the Comeonin interface](https://hexdocs.pm/comeonin/).
+
+### Forbidding access
+
+The generated code ships with an auth module with a handful of plugs that fetch the current user, require authentication and so on. For instance, in an app named Demo which had `mix phx.gen.auth Accounts User users` run on it, you will find a module named `DemoWeb.UserAuth` with plugs such as:
+
+  * `fetch_current_user` - fetches the current user information if available
+  * `require_authenticated_user` - must be invoked after `fetch_current_user` and requires that a current user exists and is authenticated
+  * `redirect_if_user_is_authenticated` - used for the few pages that must not be available to authenticated users
+
+### Confirmation
+
+The generated functionality ships with an account confirmation mechanism, where users have to confirm their account, typically by email. However, the generated code does not forbid users from using the application if their accounts have not yet been confirmed. You can trivially add this functionality by customizing the plugs generated in the Auth module.
```

Comment:
Avoid works like trivially because, whenever a user cannot pull it off, they will feel frustrated.

```suggestion
The generated functionality ships with an account confirmation mechanism, where users have to confirm their account, typically by email. However, the generated code does not forbid users from using the application if their accounts have not yet been confirmed. You can add this functionality by customizing the `require_authenticated_user` in the `Auth` module to check for the `confirmed_at` field (and any other property you desire).
```